File: Arjuna/versions/PR2.0_Alpha_2/ObjectStore/src/StoreList.cc
/*
 * Copyright (C) 1993
 *
 * Department of Computing Science,
 * The University,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: StoreList.cc,v 
 */

#ifndef DEBUG_H_
#  include <Common/Debug.h>
#endif

#ifndef STORELIST_H_
#  include <ObjectStore/StoreList.h>
#endif

StoreList::StoreList ()
                      : useCount(0),
			instance(0),
			next(0)
{
#ifdef DEBUG
    debug_stream << CONSTRUCTORS << FAC_OBJECT_STORE << VIS_PUBLIC;
    debug_stream << "StoreList::StoreList ()" << endl;
#endif    
}

StoreList::~StoreList ()
{
#ifdef DEBUG
    debug_stream << DESTRUCTORS << FAC_OBJECT_STORE << VIS_PUBLIC;    
    debug_stream << "StoreList::~StoreList ()" << endl;
#endif

    if (instance)
	delete instance;
}

File: JTSArjuna/JTSArjuna-1.2.2/JavaArjunaLite/classes/com/arjuna/JavaArjunaLite/Implementation/ObjectStore/StoreList.java
/*
 * Copyright (C) 1996, 1997, 1998,
 *
 * Department of Computing Science,
 * University of Newcastle upon Tyne,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: StoreList.java,v 1.3 1998/07/06 13:26:48 nmcl Exp $
 */

package com.arjuna.JavaArjunaLite.Implementation.ObjectStore;

import com.arjuna.JavaArjunaLite.Implementation.ObjectStoreImple;

/*
 * Default visibility.
 */

class StoreList
{

public StoreList ()
    {
	useCount = 0;
	instance = null;
	next = null;
    }

public long useCount;
public ObjectStoreImple instance;
public StoreList next;
    
}
